File: memento/src/test/java/com/alexstyl/specialdates/events/namedays/calendar/resource/RomanianEasterSpecialCalculatorTest.java
package com.alexstyl.specialdates.events.namedays.calendar.resource;

import com.alexstyl.specialdates.date.Date;
import com.alexstyl.specialdates.events.namedays.calendar.OrthodoxEasterCalculator;

import java.util.ArrayList;
import java.util.List;

import org.junit.Test;

import static org.fest.assertions.api.Assertions.assertThat;

public class RomanianEasterSpecialCalculatorTest {

    private static final OrthodoxEasterCalculator easterCalculator = new OrthodoxEasterCalculator();

    @Test
    public void calculatesSundayBeforeEasterCorrectly() {
        RomanianEasterSpecialCalculator calculator = new RomanianEasterSpecialCalculator(easterCalculator);
        List<Date> expectedDates = buildExpectedDates();
        for (Date expectedDate : expectedDates) {
            Date actualDate = calculator.calculateSpecialRomanianDayForYear(expectedDate.getYear());
            assertThat(expectedDate).isEqualTo(actualDate);
        }
    }

    private List<Date> buildExpectedDates() {
        List<Date> dates = new ArrayList<>();
        dates.add(Date.Companion.on(9, 4, 2017));
        dates.add(Date.Companion.on(1, 4, 2018));
        dates.add(Date.Companion.on(21, 4, 2019));
        dates.add(Date.Companion.on(12, 4, 2020));
        dates.add(Date.Companion.on(25, 4, 2021));
        dates.add(Date.Companion.on(17, 4, 2022));
        return dates;
    }
}

File: memento/src/test/java/com/alexstyl/specialdates/events/namedays/calendar/resource/RomanianEasterSpecialCalculatorTest.kt
package com.alexstyl.specialdates.events.namedays.calendar.resource

import com.alexstyl.specialdates.date.Date
import com.alexstyl.specialdates.date.dateOn
import com.alexstyl.specialdates.events.namedays.calendar.OrthodoxEasterCalculator
import org.fest.assertions.api.Assertions.assertThat
import org.junit.Test

class RomanianEasterSpecialCalculatorTest {

    private var orthodoxEasterCalculator = OrthodoxEasterCalculator()

    @Test
    fun calculatesSundayBeforeEasterCorrectly() {
        val calculator = RomanianEasterSpecialCalculator(orthodoxEasterCalculator)
        val expectedDates = buildExpectedDates()
        for (expectedDate in expectedDates) {
            val actualDate = calculator.calculateSpecialRomanianDayForYear(expectedDate.year!!)
            assertThat(expectedDate).isEqualTo(actualDate)
        }
    }

    private fun buildExpectedDates(): List<Date> =
            listOf(dateOn(9, 4, 2017),
                    dateOn(1, 4, 2018),
                    dateOn(21, 4, 2019),
                    dateOn(12, 4, 2020),
                    dateOn(25, 4, 2021),
                    dateOn(17, 4, 2022))
}
